tests/test_google_ads_field_exclusion_invalid.py

```diff
@@ -0,0 +1,245 @@
+    TODO Verify when given field selected, `fieldExclusions` fields in metadata are grayed out and cannot be selected (Manual case)
+    """
+
+    @staticmethod
+    def name():
+        return "tt_google_ads_field_exclusion_invalid"
+
+    def perform_exclusion_verification(self, field_exclusion_dict):
+        """
+        Verify for a pair of fields that if field_1 is in field_2's exclusion list then field_2 must be in field_1's exclusion list
+        """
+        error_dict = {}
+        for field, values in field_exclusion_dict.items():
+            if values != []:
+                for value in values:
+                    if value in field_exclusion_dict.keys():
+                        if field not in field_exclusion_dict[value]:
+                            if field not in error_dict.keys():
+                                error_dict[field] = [value]
+                            else:
+                                error_dict[field] += [value]
+
+        return error_dict
+
+    def random_field_gather(self, input_fields_with_exclusions):
+        """
+        Method takes list of fields with exclusions and generates a random set fields without conflicts as a result
+        The set of fields with exclusions is generated in random order so that different combinations of fields can
+        be tested over time.  A single invalid field is then added to violate exclusion rules.
+        """
+
+        # Build random set of fields with exclusions.  Select as many as possible
+        all_fields = input_fields_with_exclusions + self.fields_without_exclusions
+        randomly_selected_list_of_fields_with_exclusions = []
+        remaining_available_fields_with_exclusions = input_fields_with_exclusions
+        while len(remaining_available_fields_with_exclusions) > 0:
+            # Randomly select one field that has exclusions
+            newly_added_field = remaining_available_fields_with_exclusions[
+                random.randrange(len(remaining_available_fields_with_exclusions))]
+            # Save list for debug incase test fails
+            self.random_order_of_exclusion_fields[self.stream].append(newly_added_field,)
+            randomly_selected_list_of_fields_with_exclusions.append(newly_added_field)
+            # Update remaining_available_fields_with_exclusinos based on random selection
+            newly_excluded_fields_to_remove = self.field_exclusions[newly_added_field]
+            # Remove newly selected field
+            remaining_available_fields_with_exclusions.remove(newly_added_field)
+            # Remove associated excluded fields
+            for field in newly_excluded_fields_to_remove:
+                if field in remaining_available_fields_with_exclusions:
+                    remaining_available_fields_with_exclusions.remove(field)
+
+        # Now add one more exclusion field to make the selection invalid
+        found_invalid_field = False
+        while found_invalid_field == False:
```

Comment:
I think we could simplify this portion by generating one set of all tap-supported fields that would be invalid for the current selection and randomly picking one of those.